evalml/pipelines/component_graph.py

```diff
@@ -35,6 +35,26 @@ def __init__(self, component_dict=None, random_seed=0):
         self._feature_provenance = {}
         self._i = 0
 
+    @classmethod
+    def linearized_component_graph(cls, components):
+        """Return a list of (component name, component class) tuples from a pre-initialized component graph defined
+        as either a list or a dictionary. The component names are guaranteed to be unique."""
+        names = []
+        if isinstance(components, list):
+            seen = set()
+            for idx, component in enumerate(components):
+                component_class = handle_component_class(component)
+                component_name = component_class.name
+
+                if component_name in seen:
+                    component_name = f'{component_name}_{idx}'
+                seen.add(component_name)
+                names.append((component_name, component_class))
```

Comment:
If you're asking why I'm adding to the set as I go its because I need to keep track of which components have been duplicated. This logic used to exist in `from_list` method of `ComponentGraph` but now I moved it here so that it could be used by different pipeline methods and the iterative algorithm more easily.

Let me know if I'm misunderstanding!